### PR TITLE
new SCALAR_IN_ARRAY function analogous to DRUID_IN

### PR DESCRIPTION
**Pull request by sreemanamala**
### Description

creates a new function `SCALAR_IN_ARRAY(expr, arr)` to check if the scalar expr `expr` is present in the array expr `arr` or not

#### Release note

<hr>

##### Key changed/added classes in this PR
 * `ScalarInOperatorConversion.java`
 * `Function.java`

<hr>

This PR has:

- [ ] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] a release note entry in the PR description.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

File: docs/querying/math-expr.md [hunk 1]
```diff
@@ -176,24 +176,25 @@ See javadoc of java.lang.Math for detailed explanation for each function.
...
-| array_set_add_all(arr1,arr2) | combines the unique set of elements of 2 arrays, the resulting array type determined by the type of the first array |
-| array_slice(arr,start,end) | return the subarray of arr from the 0 based index start(inclusive) to end(exclusive), or `null`, if start is less than 0, greater than length of arr or less than end|
-| array_to_string(arr,str) | joins all elements of arr by the delimiter specified by str |
-| string_to_array(str1,str2) | splits str1 into an array on the delimiter specified by str2, which is a regular expression |
+| function                     | description                                                                                                                                                                                                   |
+|------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
```
Comment (kgyrtkirk) [inline docs/querying/math-expr.md:180]: I believe `.md` files are processed and rendered during page generation - is it really needed to add whitespace change to to unrelated lines - to make it a nice ascii table?
even the preious version was not caring at all if its a nice ascii or not by using `| --- | --- |`

File: processing/src/main/java/org/apache/druid/math/expr/Function.java [hunk 2]
```diff
@@ -3724,6 +3724,44 @@ ExprEval doApply(ExprEval arrayExpr, ExprEval scalarExpr)
     }
   }
 
+  class ArrayScalarInFunction extends ArrayScalarFunction
```
Comment (kgyrtkirk) [inline processing/src/main/java/org/apache/druid/math/expr/Function.java:3727]: I wonder if it would be possible to instead of adding a new native function - delegate all this work to the new type-aware native in filter @clintropolis have added?

File: processing/src/main/java/org/apache/druid/math/expr/Function.java [hunk 3]
```diff
@@ -3724,6 +3724,44 @@ ExprEval doApply(ExprEval arrayExpr, ExprEval scalarExpr)
...
+    }
+
+    @Override
+    Expr getArrayArgument(List<Expr> args)
+    {
+      return args.get(1);
+    }
+
+    @Override
+    ExprEval doApply(ExprEval arrayExpr, ExprEval scalarExpr)
```
Comment (gianm) [inline processing/src/main/java/org/apache/druid/math/expr/Function.java:3755]: You don't need to do this now, but as a follow up you could add an `asSingleThreaded` impl that creates a Set to avoid the `Arrays.asList(array).contains`, similar to the technique used in `OverlapConstantArray`.

**Review by gianm — APPROVED** [review]
